apis/v1alpha2/shared_types.go

```diff
@@ -186,14 +186,57 @@ type BackendRef struct {
 // RouteConditionType is a type of condition for a route.
 type RouteConditionType string
 
+// RouteConditionReason is a reason for a route condition.
+type RouteConditionReason string
+
 const (
 	// This condition indicates whether the route has been accepted or rejected
 	// by a Gateway, and why.
-	ConditionRouteAccepted RouteConditionType = "Accepted"
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Accepted"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	RouteConditionAccepted RouteConditionType = "Accepted"
+
+	// FIXME: alias for backwards compatibility in v1alpha2, remove in next release
+	ConditionRouteAccepted RouteConditionType = RouteConditionAccepted
 
 	// This condition indicates whether the controller was able to resolve all
 	// the object references for the Route.
-	ConditionRouteResolvedRefs RouteConditionType = "ResolvedRefs"
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "ResolvedRefs"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "RefNotPermitted"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	RouteConditionResolvedRefs RouteConditionType = "ResolvedRefs"
+
+	// FIXME: alias for backwards compatibility in v1alpha2, remove in next release
+	ConditionRouteResolvedRefs RouteConditionType = RouteConditionResolvedRefs
+
+	// This reason is used with the "Accepted" condition when the Route has been
+	// accepted by the Gateway.
+	RouteReasonAccepted RouteConditionReason = "Accepted"
```

Comment:
Nit: In other place we've grouped conditions and acceptable reasons together in type definitions. So essentially

Condition Accepted
- Reason Accepted

Condition ResolvedRefs
- Reason ResolvedRefs
- ReasonRefNotPermitted